src/analysis/convex/topology.lean

```diff
@@ -316,6 +317,30 @@ by simpa only [metric.ball, sep_univ] using (convex_on_univ_dist a).convex_lt r
 lemma convex_closed_ball (a : E) (r : ℝ) : convex ℝ (metric.closed_ball a r) :=
 by simpa only [metric.closed_ball, sep_univ] using (convex_on_univ_dist a).convex_le r
 
+lemma convex.thickening (hs : convex ℝ s) (δ : ℝ) : convex ℝ (thickening δ s) :=
+by { rw ←add_ball_zero, exact hs.add (convex_ball 0 _) }
+
+lemma convex.cthickening (hs : convex ℝ s) (δ : ℝ) : convex ℝ (cthickening δ s) :=
+begin
+  obtain hδ | hδ := le_total 0 δ,
+  { rw cthickening_eq_Inter_thickening hδ,
+    exact convex_Inter₂ (λ _ _, hs.thickening _) },
+  { rw cthickening_of_nonpos hδ,
+    exact hs.closure }
+end
+
+/-- If `s`, `t` are disjoint convex sets, `s` is compact and `t` is closed then we can find open
+disjoint convex sets containing them. -/
+-- TODO: This proof uses the normed space structure of `E`, but it could work for locally convex
+-- topological vector spaces: instead of looking at thickenings, we could show there must be some
+-- convex neighbourhood `u` of 0 which make `s + u` and `t + u` disjoint?
+lemma disjoint.exists_open_convexes (hs₁ : convex ℝ s) (hs₂ : is_compact s)
+  (ht₁ : convex ℝ t) (ht₂ : is_closed t) (disj : disjoint s t) :
```

Comment:
Could you put the `disjoint s t` assumption as the first assumption, to make sure dot notation works optimally?